packages/babel-traverse/src/scope/lib/renamer.js

```diff
@@ -2,9 +2,40 @@ import Binding from "../binding";
 import splitExportDeclaration from "@babel/helper-split-export-declaration";
 import * as t from "@babel/types";
 
+export function isPathInFunctionParameter(path) {
+  let p = path;
+  while (p != null) {
+    if (
+      p.parentPath &&
+      p.parentPath.isFunction() &&
+      p.parentPath.node.params.includes(p.node)
+    ) {
+      return true;
+    }
+    p = p.parentPath;
+  }
+  return false;
+}
+
 const renameVisitor = {
-  ReferencedIdentifier({ node }, state) {
+  ReferencedIdentifier(path, state) {
+    const { node } = path;
     if (node.name === state.oldName) {
+      let p = path;
+      while (
+        p.parentPath &&
+        !p.parentPath.isAssignmentPattern({ right: p.node }) &&
+        !p.isObjectProperty({ computed: true })
```

Comment:
We should only stop if we were inside the key, not if we were inside the "value"/pattern of a computed property.

Reply:
I reworked on #10912 to get rid of the spaghetti. 😝